json-endpoints/jetpack/class.jetpack-json-api-cron-endpoint.php

```diff
@@ -0,0 +1,228 @@
+<?php
+
+// GET /sites/%s/cron
+class Jetpack_JSON_API_Cron_Endpoint extends Jetpack_JSON_API_Endpoint {
+	protected $needed_capabilities = 'manage_options';
+
+	protected function validate_call( $_blog_id, $capability, $check_manage_active = true ) {
+		parent::validate_call( $_blog_id, $capability, false );
+	}
+
+	protected function result() {
+		return array(
+			'cron_array' => _get_cron_array(),
+			'current_timestamp' => time()
+		);
+	}
+
+	protected function sanitize_hook( $hook ) {
+		return preg_replace( '/[^A-Za-z0-9-_]/', '', $hook );
+	}
+
+	protected function resolve_arguments() {
+		$args = $this->input();
+		return  ! isset( $args['arguments'] ) ? array() : json_decode( $args['arguments'] );
+	}
+
+	protected function is_cron_locked( $gmt_time ) {
+		// The cron lock: a unix timestamp from when the cron was spawned.
+		$doing_cron_transient = $this->get_cron_lock();
+		if ( $doing_cron_transient && ( $doing_cron_transient + WP_CRON_LOCK_TIMEOUT > $gmt_time )  ) {
+			return new WP_Error( 'cron-is-locked', 'Current there is a cron already happening.' );
+		}
+		return $doing_cron_transient;
+	}
+
+	protected function maybe_unlock_cron( $doing_wp_cron ) {
+		if ( $this->get_cron_lock() == $doing_wp_cron ) {
+			delete_transient( 'doing_cron' );
+		}
+	}
+
+	protected function lock_cron() {
+		$lock = sprintf( '%.22F', microtime( true ) );
+		set_transient( 'doing_cron', $lock );
+		return $lock;
+	}
+
+	/**
+	 * This function is based on the one found in wp-cron.php with a simular name
+	 * @return int
+	 */
+	protected function get_cron_lock() {
+		global $wpdb;
+
+		$value = 0;
+		if ( wp_using_ext_object_cache() ) {
+			/*
+			 * Skip local cache and force re-fetch of doing_cron transient
+			 * in case another process updated the cache.
+			 */
+			$value = wp_cache_get( 'doing_cron', 'transient', true );
+		} else {
+			$row = $wpdb->get_row( $wpdb->prepare( "SELECT option_value FROM $wpdb->options WHERE option_name = %s LIMIT 1", '_transient_doing_cron' ) );
+			if ( is_object( $row ) ) {
+				$value = $row->option_value;
+			}
+		}
+		return $value;
+	}
+}
+
+// POST /sites/%s/cron
+class Jetpack_JSON_API_Cron_Post_Endpoint extends Jetpack_JSON_API_Cron_Endpoint {
+
+	protected function result() {
+		define('DOING_CRON', true);
+
+		$args = $this->input();
+
+		if ( false === $crons = _get_cron_array() ) {
+			return new WP_Error( 'no-cron-event', 'Current there are no cron events' );
```

Comment:
I would add an "ly" to "current" to make the string "Currently there are no cron events".